### PR TITLE
community[patch]: Fix HuggingFace LLM to not repeat the prompt as part of the result

### PR DESCRIPTION
**Pull request by lin-calvin**
Description: Let huggingface llm don't repeat prompt
Issue: #16972 
Break changes: It changes the llm `Huggingface`'s behavior:
  - Before: Huggingface llm will repeat the prompt in its result
  - After:    Huggingface llm will not repeat the prompt in its result

### CODE REVIEW

**Comment by eyurtsev** [comment]
@calvinweb this looks like a breaking change, but also a fix for incorrect behavior. Any chance you could include a pr summary that explains the before and after of this change?

File: libs/community/langchain_community/llms/huggingface_hub.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ def _call(
                 response = hf("Tell me a joke.")
         """
         _model_kwargs = self.model_kwargs or {}
-        parameters = {**_model_kwargs, **kwargs}
+        parameters = {**_model_kwargs, **kwargs, "return_full_text": False}
```
Comment (mspronesti) [inline libs/community/langchain_community/llms/huggingface_hub.py:130]: Isn't the current version more versatile? The user can pass `return_full_text` as a kwarg or as a model_kwarg to force this behavior if they need it.

Comment (calvinweb) [reply libs/community/langchain_community/llms/huggingface_hub.py:130]: return_full_text should be the default to let it behave like other **langchain** language models

File: libs/community/langchain_community/llms/huggingface_hub.py [hunk 2]
```diff
@@ -127,7 +127,7 @@ def _call(
                 response = hf("Tell me a joke.")
         """
         _model_kwargs = self.model_kwargs or {}
-        parameters = {**_model_kwargs, **kwargs}
+        parameters = {"return_full_text": False**_model_kwargs, **kwargs, "return_full_text": False}
```
Comment (mspronesti) [inline libs/community/langchain_community/llms/huggingface_hub.py:130]: Making it the first parameter looks better to me. Please notice that you have one syntax error and a typo in your last commit: you are missing a comma after the first item of the dictionary and you have `return_full_text` twice.

**Comment by calvinweb** [comment]
> @calvinweb this looks like a breaking change, but also a fix for incorrect behavior. Any chance you could include a pr summary that explains the before and after of this change?

It is now included in the comment

File: libs/community/langchain_community/llms/huggingface_hub.py [hunk 3]
```diff
@@ -127,7 +127,7 @@ def _call(
                 response = hf("Tell me a joke.")
         """
         _model_kwargs = self.model_kwargs or {}
-        parameters = {**_model_kwargs, **kwargs}
+        parameters = {"return_full_text": False, **_model_kwargs, **kwargs, "return_full_text": False}
```
Comment (eyurtsev) [inline libs/community/langchain_community/llms/huggingface_hub.py:130]: Fixing typo
```suggestion
        parameters = {"return_full_text": False, **_model_kwargs, **kwargs}
```

**Comment by eyurtsev** [comment]
Fix typo -- parameter was being passed twice. Ready to merge as long as unit-tests pass

**Review by eyurtsev — APPROVED** [review]

File: libs/community/langchain_community/llms/huggingface_hub.py [hunk 4]
```diff
@@ -127,7 +127,7 @@ def _call(
                 response = hf("Tell me a joke.")
         """
         _model_kwargs = self.model_kwargs or {}
-        parameters = {**_model_kwargs, **kwargs}
+        parameters = {"return_full_text": False, **_model_kwargs, **kwargs}
```
Comment (baskaryan) [inline libs/community/langchain_community/llms/huggingface_hub.py:130]: do all models support this parameter?

**Comment by baskaryan** [comment]
cc @aymeric-roucher believe #17254 also addresses this correct?

**Comment by calvinweb** [comment]
> cc @aymeric-roucher believe #17254 also addresses this correct?

Since it is merged, should this issue close?

**Comment by eyurtsev** [comment]
Closing since PR was addressed already

**Pull request closed without merging**